### PR TITLE
Show multiplied grid cell index

### PR DESCRIPTION
**Pull request by Liebranca**
Implements #2082. The cell index is displayed in the same way the tile index is shown for a tilemap layer, as both values serve more or less the same purpose.

To try and avoid confusion, I've made it so the cell index only appears if the active layer is not a tilemap, else the tile index logic is used. I'd like some feedback on this as I'm not entirely sure how obvious this would be from the user's perspective.

Perhaps it'd be more clear if both values are displayed? I thought of using `()` parentheses for the cell index rather than `[]` brackets, so as to differentiate them. Maybe this could be done only when both indices are displayed.

### CODE REVIEW

File: src/app/ui/editor/standby_state.cpp [hunk 1]
```diff
@@ -614,6 +614,21 @@ bool StandbyState::onUpdateStatusBar(Editor* editor)
...
+        else if ((!site.layer() ||
+            !site.layer()->isTilemap()) &&
+            (spritePos.x >= 0 && spritePos.x <= sprite->width() &&
+            spritePos.y >= 0 && spritePos.y <= sprite->height())) {
+          int columns = int(std::floor(
+            sprite->bounds().w/grid.tileSize().w));
+          int rows = int(std::floor(
+            sprite->bounds().h/grid.tileSize().h));
+          int column = pt.x%columns;
+          int row = pt.y%rows;
```
Comment (dacap) [inline src/app/ui/editor/standby_state.cpp:627]: I'd prefer to check for columns != 0 and rows != 0 before doing this.

File: src/app/ui/editor/standby_state.cpp [hunk 2]
```diff
@@ -614,6 +614,21 @@ bool StandbyState::onUpdateStatusBar(Editor* editor)
             buf += fmt::format(" [{}{}]", ti, str);
           }
         }
+        // Show the grid cell index
+        else if ((!site.layer() ||
+            !site.layer()->isTilemap()) &&
+            (spritePos.x >= 0 && spritePos.x <= sprite->width() &&
+            spritePos.y >= 0 && spritePos.y <= sprite->height())) {
```
Comment (dacap) [inline src/app/ui/editor/standby_state.cpp:621]: We can use
```suggestion
            sprite->bounds().contains(spritePos)) {
```

**Review by dacap — CHANGES_REQUESTED** [review]
Sorry for the late testing about this PR @Liebranca. As you said I think it might be confusing using the same notation `[]` for two different purposes: in tilemaps it shows the tile index, and with this PR in normal layers it would shows the grid tile index in a linear array.

I also think that this field might probably be most useful in tilemap layers, so it should appear in all kind of layers. Not sure about the notation, but maybe showing something like:

```c++
          buf += fmt::format(" :search: {}", column+row*columns);
```

And I'm not sure what we should do with partially visible grid tiles:

https://github.com/user-attachments/assets/27421cc6-d4b7-42fe-933b-1a07eb9a328d

**Comment by Liebranca** [comment]
Hi @dacap! About partially visible cells, I purposely made it so the index wraps back to the beggining or end of the row/column. That way a cell being "cut" by grid offset gives the same index as the one on the opposite side, as if the missing portion of the tile was continued there.

However, this may not make that much sense after all. A possible alternative I'm testing right now is considering such partially visible cells as individual tiles, rather than portions of a single tile spread across two to four corners of the screen.

**Pull request closed without merging**